### PR TITLE
feat: Added shared vpc module

### PR DESCRIPTION
**Pull request by Kkhandale2024**
## what
* Added shared vpc module
image

### CODE REVIEW

**Comment by VishwajitNagulkar** [comment]
Review and update all workflows `.github/workflows`  and use our shared workflow

File: _example/variables.tf [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#-------------------------------------------------------------------------------
+                      # Google_Provider_variables #
+#-------------------------------------------------------------------------------
+
 variable "environment" {
```
Comment (Tanveer143s) [inline _example/variables.tf:5]: add description

File: variables.tf [hunk 2]
```diff
@@ -89,3 +96,29 @@ variable "module_depends_on" {
   description = "(Optional) A list of external resources the module depends_on. Default is '[]'."
 
 }
+
+#---------------------------------------------------------------------------------------------------------
+                        # shared_vpc_variable #
+#----------------------------------------------------------------------------------------------------------
+
+variable "host_project_id" {
```
Comment (Tanveer143s) [inline variables.tf:104]: set as per standards

**Review by Tanveer143s — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 3]
```diff
@@ -89,3 +96,29 @@ variable "module_depends_on" {
...
+  description = "Google Cloud Project ID"
+}
+
+variable "service_project_id" {
+  description = "Project ID of the service project"
+}
```
Comment (Tanveer143s) [inline variables.tf:110]: set as per standards like (type, default) missing

**Review by Tanveer143s — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 4]
```diff
@@ -23,6 +26,10 @@ variable "name" {
 
 }
 
+#---------------------------------------------------------------------------------------------------
+                # Google_Compute_Network_(VPC)_variable #
+#---------------------------------------------------------------------------------------------------
+
 variable "google_compute_network_enabled" {
   type    = bool
   default = true
```
Comment (Tanveer143s) [inline variables.tf:35]: add descriptions here

File: _example/variables.tf (hunk 1 above)
Comment (Kkhandale2024) [reply _example/variables.tf:5]: Done sir

File: variables.tf (hunk 3 above)
Comment (Kkhandale2024) [reply variables.tf:115]: Done sir

File: variables.tf (hunk 4 above)
Comment (Kkhandale2024) [reply variables.tf]: done sir added a description

File: .github/dependabot.yml [hunk 5]
```diff
@@ -5,12 +5,52 @@
...
+      - "clouddrove-ci"
+    # Add reviewer
+    reviewers:
+      - "approvers"
+    # Allow up to 3 open pull requests for pip dependencies
+    open-pull-requests-limit: 3
+
   - package-ecosystem: "terraform" # See documentation for possible values
-    directory: "_example/" # Location of package manifests
+    directory: "/_example/complete" # Location of package manifests
```
Comment (VishwajitNagulkar) [inline .github/dependabot.yml:46]: Update your example directory path

File: .github/workflows/tf-checks.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  workflow_dispatch:
+jobs:
+  basic:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.2
+    with:
+      working_directory: './_example/'
+  complete:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.2
+    with:
+      working_directory: './_example/'
```
Comment (VishwajitNagulkar) [inline .github/workflows/tf-checks.yml:15]: we have one example only so update according to it

File: .github/dependabot.yml (hunk 5 above)
Comment (Kkhandale2024) [reply .github/dependabot.yml]: Done sir

File: .github/workflows/tf-checks.yml (hunk 6 above)
Comment (Kkhandale2024) [reply .github/workflows/tf-checks.yml]: Sir I have done a changes please check if there are any mistake please let me know

**Review by VishwajitNagulkar — APPROVED** [review]
LGTM

**Review by AshutoshAM2002 — APPROVED** [review]
LGTM

**Pull request merged**